### ISSUE TITLE
"__fake__" appears on LHN after flagging a content

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#20940]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Start a new conversation from account A to account B (There should be no comment before the comment that is about to be flagged)
2. Flag the message sent from account B as either "harassment" or "Assault"
3. Observe the LHN

## Expected Result: 
"__fake__" should not appear on the LHN
## Actual Result: 
"__fake__" appears on LHN

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.29-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/1b10afc7-9f30-4a66-adc9-8e5a03f92e6f

https://github.com/Expensify/App/assets/43996225/21570101-bfa8-4bc8-b9a9-7ca7877bd2b8

**Expensify/Expensify Issue URL:**
**Issue reported by:** @nathan-mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686941708297639

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by aldo-expensify** [comment]
I think this fixes it: https://github.com/Expensify/App/pull/20949

**Comment by sakluger** [comment]
Thanks for the quick work Aldo. Let me know if I can help in any way.

**Comment by aldo-expensify** [comment]
During the PR https://github.com/Expensify/App/pull/20949 review we found quite a few other cases that were not working, see for more information:

- Missing offline behaviour: https://github.com/Expensify/App/pull/20949#issuecomment-1595474711
- Groups/Tasks not working correctly: https://github.com/Expensify/App/pull/20949#issuecomment-1595483433, https://github.com/Expensify/App/pull/20949#issuecomment-1595486497
- Wrong comment author for last message (LHN): https://github.com/Expensify/App/pull/20949#issuecomment-1597612599
- Wrong attachment last message: https://github.com/Expensify/App/pull/20949#issuecomment-1597674629
- Flagged comment shows in LHN if you don't open the report loading report actions: https://github.com/Expensify/App/pull/20949#issuecomment-1597691483

Slack conversation: https://expensify.slack.com/archives/C03TQ48KC/p1687203338251969

Current plan to fix it: https://expensify.slack.com/archives/C07J32337/p1687204905234929?thread_ts=1687190085.248999&cid=C07J32337

**Comment by Beamanator** [comment]
FYI I also see `__fake__` appear in LHN after creating a new group with a user I don't know yet / doesn't have a real account yet
- Found while testing steps from https://github.com/Expensify/App/issues/21075

**Comment by luacmartins** [comment]
Gonna remove the blocker label as per [this comment](https://expensify.slack.com/archives/C07J32337/p1687295606673249?thread_ts=1687190085.248999&cid=C07J32337)

**Comment by aldo-expensify** [comment]
To finish this, the following PR's will be needed:

- Auth: https://github.com/Expensify/Auth/pull/8170
- Web: https://github.com/Expensify/Web-Expensify/pull/37910
- App: https://github.com/Expensify/App/pull/20949

**Comment by aldo-expensify** [comment]
Made backend PRs ready for review.

**Comment by aldo-expensify** [comment]
Waiting for the timings here to start merging: https://github.com/Expensify/Expensify/issues/294240

**Comment by luacmartins** [comment]
@aldo-expensify just beware that we're on a new feature freeze and we shouldn't merge anything risky until Jun 29th

**Comment by aldo-expensify** [comment]
ok!, thanks for reminding me :)

**Comment by sakluger** [comment]
Not overdue, just on hold for the feature freeze.

**Comment by aldo-expensify** [comment]
I merged the Auth PR, after it gets deployed, I'll merge Web-E, and after Web-E gets deployed, we can tell the C+ to retest the App PR.

**Comment by lschurr** [comment]
I believe this will also be fixed as a result of the PR: https://github.com/Expensify/App/issues/22386

**Comment by aldo-expensify** [comment]
I just update the Web-E PR and made it ready for review again: https://github.com/Expensify/Web-Expensify/pull/37910

> I believe this will also be fixed as a result of the PR: https://github.com/Expensify/App/issues/22386

Maybe! I think we can check after the PRs for this issue have been merged

**Comment by aldo-expensify** [comment]
> Hey guys, this issue has been fixed but payment has not been proceed for it. @lschurr @aldo-expensify @sakluger

True, I retested and this is working now:

<img width="1736" alt="image" src="https://github.com/Expensify/App/assets/87341702/bdcb524d-79bc-45aa-b1aa-083a0e52386c">

**Comment by aldo-expensify** [comment]
@sakluger we have to handle the reporting bonus for @Nathan-Mulugeta

**Comment by Nathan-Mulugeta** [comment]
Bump @sakluger

**Comment by Nathan-Mulugeta** [comment]
Bump @sakluger again 😄

**Comment by sakluger** [comment]
Sorry for the delay, just sent an offer to @Nathan-Mulugeta on Upwork.

**Comment by sakluger** [comment]
Switching to a daily to reflect the priority.

**Comment by Nathan-Mulugeta** [comment]
Thanks @sakluger , just accepted the offer.

**Comment by sakluger** [comment]
All paid up! Sorry again for the delay on payment @Nathan-Mulugeta.
